File: android/app/src/main/kotlin/com/thecityandthebike/ui/screens/ScoreRulesScreen.kt
package com.thecityandthebike.ui.screens

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.thecityandthebike.data.model.ScoreRules

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ScoreRulesScreen(onBack: () -> Unit) {
    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Score Rules") },
                navigationIcon = {
                    IconButton(onClick = onBack) {
                        Icon(
                            imageVector = Icons.AutoMirrored.Filled.ArrowBack,
                            contentDescription = "Back"
                        )
                    }
                }
            )
        }
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .padding(paddingValues)
                .padding(16.dp)
                .verticalScroll(rememberScrollState())
        ) {
            Text(
                text = "Photo Points",
                style = MaterialTheme.typography.headlineSmall
            )
            Spacer(modifier = Modifier.height(8.dp))
            ScoreRules.photoRules.forEach { rule ->
                ScoreRuleRow(rule.label, rule.points)
            }
            Spacer(modifier = Modifier.height(8.dp))
            Text(
                text = "A single photo can trigger several bonuses at once \u2014 the points add up!",
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )

            Spacer(modifier = Modifier.height(24.dp))

            Text(
                text = "Tag Points",
                style = MaterialTheme.typography.headlineSmall
            )
            Spacer(modifier = Modifier.height(8.dp))
            ScoreRules.tagRules.forEach { rule ->
                ScoreRuleRow(rule.label, rule.points)
            }
        }
    }
}

@Composable
private fun ScoreRuleRow(label: String, points: Int) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 4.dp)
    ) {
        Text(
            text = label,
            style = MaterialTheme.typography.bodyLarge,
            modifier = Modifier.weight(1f)
        )
        Text(
            text = "+$points",
            style = MaterialTheme.typography.bodyLarge,
            fontWeight = FontWeight.Bold
        )
    }
}

File: ios/Packages/TCATBProfile/Sources/TCATBProfile/ScoreRulesScreen.swift
import SwiftUI
import TCATBModels

public struct ScoreRulesScreen: View {
    var onBack: () -> Void

    public init(onBack: @escaping () -> Void) {
        self.onBack = onBack
    }

    public var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 0) {
                Text("Photo Points")
                    .font(.title2)
                    .padding(.bottom, 8)

                ForEach(ScoreRules.photoRules, id: \.id) { rule in
                    ScoreRuleRow(label: rule.label, points: rule.points)
                }

                Text("A single photo can trigger several bonuses at once \u{2014} the points add up!")
                    .font(.subheadline)
                    .foregroundStyle(.secondary)
                    .padding(.top, 8)

                Text("Tag Points")
                    .font(.title2)
                    .padding(.top, 24)
                    .padding(.bottom, 8)

                ForEach(ScoreRules.tagRules, id: \.id) { rule in
                    ScoreRuleRow(label: rule.label, points: rule.points)
                }
            }
            .padding()
        }
        .navigationTitle("Score Rules")
        .navigationBarTitleDisplayMode(.inline)
        .navigationBarBackButtonHidden(true)
        .toolbar {
            ToolbarItem(placement: .navigationBarLeading) {
                Button(action: onBack) {
                    Image(systemName: "chevron.left")
                }
            }
        }
    }
}

private struct ScoreRuleRow: View {
    let label: String
    let points: Int

    var body: some View {
        HStack {
            Text(label)
                .font(.body)

            Spacer()

            Text("+\(points)")
                .font(.body)
                .fontWeight(.bold)
        }
        .padding(.vertical, 4)
    }
}
